drivers/flash/flash_shell.c

```diff
@@ -23,25 +23,37 @@ static u8_t test_arr[TEST_ARR_SIZE];
 static int cmd_erase(const struct shell *shell, size_t argc, char *argv[])
 {
 	struct device *flash_dev;
+	size_t fa_start;
 	u32_t page_addr;
+	char *endptr;
 	int result;
 	u32_t size;
 
-	flash_dev = device_get_binding(DT_FLASH_DEV_NAME);
-	if (!flash_dev) {
-		shell_error(shell, "Flash driver was not found!");
-		return -ENODEV;
-	}
-
 	if (argc < 2) {
 		shell_error(shell, "Missing page address.");
 		return -EINVAL;
 	}
+	page_addr = strtoul(argv[1], &endptr, 16);
```

Comment:
consider making a function from all this code duplications in the commands. e.g. function which gets device and pointer to arguments and prints error if device is not found. that should make those command look nicer.

Reply:
you can even add function for read, write, test which additionally gets the address

Reply:
ok, reworked to use a helper function to parse device and address argument